### DISCUSSION TITLE
VS Code Raspberry PI

### DISCUSSION THREAD

**Discussion by isidorn** [microsoft/vscode-discussions#2379] [Announcements]
# Visual Studio Code on Raspberry Pi

Even though it is not officialy supported you can run Visual Studio Code on [Raspberry Pi](https://www.raspberrypi.org) devices.

![RPi-Logo-Landscape-Reg-SCREEN](https://github.com/user-attachments/assets/bb9393b5-e987-4764-9188-e84b1bd66de6)

By downloading and using Visual Studio Code, you agree to the [license terms](https://code.visualstudio.com/license) and [privacy statement](https://go.microsoft.com/fwlink/?LinkID=528096&clcid=0x409).

## Installation

Visual Studio Code is officially distributed via the [Raspberry Pi OS](https://www.raspberrypi.org/software/operating-systems) (previously called Raspbian) APT repository, in both 32-bit and 64-bit variants.

You can install it by running:

```bash
sudo apt update
sudo apt install code
```

### Running VS Code

After installing the VS Code package, you can run VS Code by typing `code` in a terminal or launching it via the **Programming** menu.

![vscode-under-programming](https://github.com/user-attachments/assets/0c905b85-a23f-4344-b421-1d5c3b4cb5e2)

## Updates

Your Raspberry Pi should handle updating VS Code in the same way as other packages on the system:

```bash
sudo apt update
sudo apt upgrade code
```

You can always check when a new release is available in our [Updates](/updates) page.

## System requirements

VS Code is supported on these Raspberry Pi models running a 32-bit or 64-bit version of Raspberry Pi OS:

* Raspberry Pi 3 Model B/B+
* Raspberry Pi 4 Model B
* Raspberry Pi 400

While 1 GB of memory (RAM) meets the minimum system requirements, users will benefit from installing VS Code on a Raspberry Pi 4 with more memory.

First-generation Raspberry Pi modules and Raspberry Pi Zero are not supported as they only include an ARMv6 CPU.

### Workaround for poor performance

VS Code on Raspberry Pi 4 may be slow with the default setup. A workaround is to disable hardware (GPU) acceleration in VS Code:

1. Open the VS Code `argv.json` file using the **Preferences: Configure Runtime Arguments** command.
2. Set `"disable-hardware-acceleration": true`.
3. Restart VS Code.

The `"disable-hardware-acceleration": true` runtime argument switch has the effect of passing the `--disable-gpu` command-line argument on VS Code startup.

**Comment by isidorn** [comment]
Since VS Code does not officially support Raspberry PI - we need your help to comment on this issue and tell us on what Raspberry PI platforms VS Code does work. And share your experiences. Thank you

**Reply by saikrishnan107** [reply]
Why does visual studio code is soo slow and crashing every time in raspberry pi 5 even though it has a 8gb ram?

**Reply by AllyTechEngineering** [reply]
Check the version of VS Code. I rolled back to 1.96, and all is well. I posted on LinkedIn how to do it if anyone needs it. I am not sure if the issues have been corrected in the later versions.

**Reply by ptiszai** [reply]
I am fixid, this:
code --disable-gpu  --js-flags="--no-decommit-pooled-pages"

**Reply by saikrishnan107** [reply]
Can you send the linkedin post here! thank you in advance 😄

**Reply by AllyTechEngineering** [reply]
https://www.linkedin.com/posts/bob-taylor-mscs-mba_today-i-updated-the-vs-code-on-the-raspberry-activity-7293732657589362688-nNc9?utm_source=share&utm_medium=member_desktop&rcm=ACoAAAXSkzMBwTsYho28of7bTCtAG9HjpwmKy_8

**Reply by saikrishnan107** [reply]
@AllyTechEngineering hello, I have installed vscode by your instructions and disabled the gpu as well, but the app freezes all the time i open. 

I still get the issue :/

**Reply by ptiszai** [reply]
I not used apt-get installed. I downloaded "code_1.98.2-1741787947_arm64.deb" file from:
https://code.visualstudio.com/Download#
I installed vscode, after i not call if from icon of desktop, only inside the bash sctipt,.
Very important the "--js-flags="--no-decommit-pooled-pages"" option flag:
code --disable-gpu --js-flags="--no-decommit-pooled-pages"

**Comment by LuvbCa** [comment]
so tbh don't try running vscode or vscode server on a raspberry pi with 1gb ram. i am using the RPI 3B+ with 1gb ram and anything can be considered a headache on this low of ram. Sometimes it crashes or freezes up. A hard reset is the only option regarding this freezing but if you are patient enough you can use the vscode server, which works fine if you dont install too many addons.

**Comment by burrowj** [comment]
I'll wait till you guys have tested it on Pi5 and say its supported.  Any idea when that might be

**Reply by isidorn** [reply]
This page is community supported. So you should go ahead and test it if you can and report back so we all benefit.

**Reply by burrowj** [reply]
I work on a project that spans Xamarin / Maui / Blazor and now Raspberry Pi
If I have useful information I will definitely pass it on

**Reply by ogboot** [reply]
> This page is community supported. So you should go ahead and test it if you can and report back so we all benefit.

I'm building out an 8gb pi5 with vscode soon, I'll report back. Ubuntu ok?

**Reply by ogboot** [reply]
I deployed Ubuntu 24.10 on my 8gb rpi 5 & vscode runs great. It's a fun little python coding machine.
Also deployed vscode's ssh server packages on it to code remotely & that also works with no issues so far.

**Reply by AllyTechEngineering** [reply]
Worked with VS Code on my Model 5 with 8G RAM for a week and then updated my system and the VS Code is now 1.97.0 and I am running Debian bookworm (Raspberry Pi Desktop OS). VS Code opens the windows and then after a while locks up an I get the error box that it is not responding - close, wait or reopen screen. Anyone run into this?

**Reply by ogboot** [reply]
Patched Ubuntu 24.10 and updated vscode to 1.97 today on the 8gb rpi5, no issues.

**Reply by ogboot** [reply]
> Worked with VS Code on my Model 5 with 8G RAM for a week and then updated my system and the VS Code is now 1.97.0 and I am running Debian bookworm (Raspberry Pi Desktop OS). VS Code opens the windows and then after a while locks up an I get the error box that it is not responding - close, wait or reopen screen. Anyone run into this?

Sounds like an extension issue

**Reply by ptiszai** [reply]
My Model is 5 with 8G RAM,
Debian bookworm (Raspberry Pi Desktop OS).
Upgrated code 1.98.1 
Error:
The window terminated unexpectedy (reason: 'crashed', code: '5')

**Reply by gjsjohnmurray** [reply]
https://github.com/microsoft/vscode/issues/242742#issuecomment-2721797749

Fix will come in 1.98.2

**Comment by arthur-adriansens** [comment]
Hi! The Raspberry Pi 2W is amazing and works wonderful with VS Code.

You can read [This article](https://www.raspberrypi.com/news/raspberry-pi-zero-2-w-vs-code-remote-awesome/) from the official Raspberry News page for more info. 

You can view [this pull request](https://github.com/microsoft/vscode-docs/pull/7692) for more info.

**Comment by ErrinDev** [comment]
As you might expect, works well with the Pi 4b 8GB ram, I am running Legacy OS 64 bit ;).

**Comment by burrowj** [comment]
> # Visual Studio Code on Raspberry Pi
> 
> Even though it is not officialy supported you can run Visual Studio Code on [Raspberry Pi](https://www.raspberrypi.org) devices.
> 
> ![RPi-Logo-Landscape-Reg-SCREEN](https://private-user-images.githubusercontent.com/1926584/382700469-bb9393b5-e987-4764-9188-e84b1bd66de6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4NzAwMzIsIm5iZiI6MTczNTg2OTczMiwicGF0aCI6Ii8xOTI2NTg0LzM4MjcwMDQ2OS1iYjkzOTNiNS1lOTg3LTQ3NjQtOTE4OC1lODRiMWJkNjZkZTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMDNUMDIwMjEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjA5OTAyMmI0YzM3NGIyZmYxZTY5NDQ2YjY4ZWIzODMxMDRhOWU0YjI0YzEzNjVjZmI5ODU3NTI2MDg3NjdlMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.74oD3oWDdKBrkXNL45RGp2mbn8t0H-u6BaKQGcpzztc)
> 
> By downloading and using Visual Studio Code, you agree to the [license terms](https://code.visualstudio.com/license) and [privacy statement](https://go.microsoft.com/fwlink/?LinkID=528096&clcid=0x409).
> ## Installation
> 
> Visual Studio Code is officially distributed via the [Raspberry Pi OS](https://www.raspberrypi.org/software/operating-systems) (previously called Raspbian) APT repository, in both 32-bit and 64-bit variants.
> 
> You can install it by running:
> 
> ```shell
> sudo apt update
> sudo apt install code
> ```
> 
> ### Running VS Code
> 
> After installing the VS Code package, you can run VS Code by typing `code` in a terminal or launching it via the **Programming** menu.
> 
> ![vscode-under-programming](https://private-user-images.githubusercontent.com/1926584/382700541-0c905b85-a23f-4344-b421-1d5c3b4cb5e2.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4NzAwMzIsIm5iZiI6MTczNTg2OTczMiwicGF0aCI6Ii8xOTI2NTg0LzM4MjcwMDU0MS0wYzkwNWI4NS1hMjNmLTQzNDQtYjQyMS0xZDVjM2I0Y2I1ZTIuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMDNUMDIwMjEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2Y3OWY3ZWNiZTY1ZWQ4ZDBmYjIyZTNlOTczMTBmYWM4ZGFmOTI1N2UyNjI0NjZjZTNiNGRjOGEwMzBiNDNhOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.gYyQ1z7iw0xa0sag3GlCvedz6ApH6QdnIa8dFXWG1Qw)
> ## Updates
> 
> Your Raspberry Pi should handle updating VS Code in the same way as other packages on the system:
> 
> ```shell
> sudo apt update
> sudo apt upgrade code
> ```
> 
> You can always check when a new release is available in our [Updates](/updates) page.
> ## System requirements
> 
> VS Code is supported on these Raspberry Pi models running a 32-bit or 64-bit version of Raspberry Pi OS:
> 
>     * Raspberry Pi 3 Model B/B+
> 
>     * Raspberry Pi 4 Model B
> 
>     * Raspberry Pi 400
> 
> 
> While 1 GB of memory (RAM) meets the minimum system requirements, users will benefit from installing VS Code on a Raspberry Pi 4 with more memory.
> 
> First-generation Raspberry Pi modules and Raspberry Pi Zero are not supported as they only include an ARMv6 CPU.
> ### Workaround for poor performance
> 
> VS Code on Raspberry Pi 4 may be slow with the default setup. A workaround is to disable hardware (GPU) acceleration in VS Code:
> 
>     1. Open the VS Code `argv.json` file using the **Preferences: Configure Runtime Arguments** command.
> 
>     2. Set `"disable-hardware-acceleration": true`.
> 
>     3. Restart VS Code.
> 
> 
> The `"disable-hardware-acceleration": true` runtime argument switch has the effect of passing the `--disable-gpu` command-line argument on VS Code startup.

I am using a Raspberry Pi 4 (2GB RAM)  and its running very slow. It to do this 

     1. Open the VS Code `argv.json` file using the **Preferences: Configure Runtime Arguments** command.

when you say "command", is this a terminal command line statement?

**Reply by gjsjohnmurray** [reply]
By "command" it means the VS Code Command Palette, which you can usually access by typing Ctrl+Shift+P or alternatively by pressing F1.

**Reply by burrowj** [reply]
Got it, thanks for that

**Comment by burrowj** [comment]
So I have the following setup
Raspberry Pi 4 with 2GB Memory
Running Signal K server on Node.js
Chromium browser  - Open with two tabs
VS Code (with GPU `"disable-hardware-acceleration": true`) - Open

I can understand this might be pushing it for this setup, but 

After a while of using the device in this config, i go to Chromium and do a search, everything comes a halt.  After 10 - 20 seconds I get control back.  If I close VS Code then everything works fine again.  I can then open VS Code again and it will work fine for a while and then this issue will arise again.  Is this your experience as well?  I was just wondering if there are any other settings in VS Code that I can configure to help this situation.

Thanks

**Comment by SmartInteractiveGroup** [comment]
Hi everyone, 

And what about the new, shiny Raspberry Pi 5 ?  https://www.raspberrypi.com/news/16gb-raspberry-pi-5-on-sale-now-at-120/  will it work? Visual Studio Code (for python programming) will work?

**Reply by ogboot** [reply]
Yes I have tested a new rpi5 with a mere 8gb & vscode works great.

**Reply by AllyTechEngineering** [reply]
Worked with VS Code on my Model 5 with 8G RAM for a week and then updated my system and the VS Code is now 1.97.0 and I am running Debian bookworm (Raspberry Pi Desktop OS). VS Code opens the windows and then after a while locks up an I get the error box that it is not responding - close, wait or reopen screen. Anyone run into this?

**Reply by gjsjohnmurray** [reply]
Maybe https://github.com/microsoft/vscode/issues/239838

**Reply by ogboot** [reply]
Actually I'm also having this issue, didn't notice initially in a Jupyter notebook.

**Reply by ogboot** [reply]
Rolled back to 1.96 for now until Microsoft gets it sorted out.

**Reply by t98nguye** [reply]
I have the issue as AllyTechEngineering described above. Any suggesting solution? Thank you

**Reply by ogboot** [reply]
Disable gpu acceleration & await an update from M$

**Reply by AllyTechEngineering** [reply]
I rolled back to 1.96, too, and all is well. I posted on LinkedIn how to do it if anyone needs it.

**Reply by t98nguye** [reply]
Thank you. Yes, I also rolled back to 1.96 and it seems working fine for now.

**Reply by ogboot** [reply]
This is fixed in 1.97.2, I've updated & confirm that the issue is resolved.

**Reply by juniorHn89** [reply]
hola! yo lo instale en mi raspberry pi 5 y no me va, Error! Crashed 5 es que lo me dice y que cierre la ventana y abrar una nueva pero vuelve a pasar lo mismo.....?

**Reply by gjsjohnmurray** [reply]
https://github.com/microsoft/vscode/issues/242742#issuecomment-2732924475 until you get to 1.99

**Comment by matteo-mosca** [comment]
It's working fine on the Pi 5. My real question is: how to get intellisense for c# working? The c# dev kit exension says "not supported" and cannot be installed.
